### PR TITLE
External provider impl

### PR DESCRIPTION
**Pull request by ItsLuized**
This pull request aims to implement the RPChProvider as an External Provider type in order to use it with wallets that use Web3Provider type providers.

In Frame wallet they use their own provider called `eth-provider`, this provider is more similar to Web3Provider instead of JsonRpcProvider. In order to implement RPCh in this wallet we need to create a Provider type more similar to Web3Provider to use the same methods and return types they use.

### CODE REVIEW

**Comment by ItsLuized** [comment]
Unit tests for this ExternalProvider type are still missing

**Comment by ItsLuized** [comment]
To inject the provider in Frame, it must be created in main.chains/index.js in the function _createProvider(). 
https://github.com/Rpc-h/extension-frame/blob/develop/main/chains/index.js#LL65C1-L66C1

No blockers until the moment.

**Comment by diegoalzate** [comment]
@nionis Not sure if the new provider should be moved to a separate file, or we rename ethers package to providers

**Comment by nionis** [comment]
@diegoalzate it should be a new adaptor in `packages`

**Review by nionis — CHANGES_REQUESTED** [review]
README references wrong ethereum-provider GH project

**Comment by diegoalzate** [comment]
Adaptor now only does what EIP 1993 states. removing any custom behaviour from specific wallets.

**Review by nionis — APPROVED** [review]
please resolve conflicts and merge

File: packages/ethereum-provider/README.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+const PROVIDER_URL = 'https://primary.gnosis-chain.rpc.hoprtech.net';
+const TIMEOUT = 10000;
+const DISCOVERY_PLATFORM_API_ENDPOINT = 'https://staging.discovery.rpch.tech';
+
+const provider = new RPChEthereumProvider(
+  PROVIDER_URL,
+  {
+    crypto: RPChCrypto,
+    client: '',
```
Comment (nionis) [inline packages/ethereum-provider/README.md:26]: add same `client` text as in the ethers README

**Pull request merged**